### ISSUE TITLE
Make uvicorn optional

### ISSUE THREAD

**Issue by davidbrochart** [jupyter-server/fps#25]
### Problem

Currently, uvicorn is a dependency of FPS.

### Proposed Solution

uvicorn should be an optional dependency, as FPS could be backed by any ASGI server.
Also, this would make it possible to use jupyverse in JupyterLite.

**Comment by adriendelsalle** [comment]
Hey thanks @davidbrochart for opening this issue.

I would rephrase a bit the problem because it was a design choice from the beginning to make `uvicorn` the only available server.
Do please update your comment if your main issue is to make `FPS` usable directly with `JupyterLite` by keeping it a pure python project.

**Comment by davidbrochart** [comment]
I guess we would really be interested in both, taking advantage of the ASGI standard to be able to swap the server (and use another one than uvicorn), or mock the server (and plug JupyterLite at the ASGI level).
What does make FPS so tightly coupled with uvicorn?

**Comment by davidbrochart** [comment]
I'd say the server agnosticism is kind of guaranteed by the ASGI standard.

**Comment by adriendelsalle** [comment]
That part yes, the logger implementation is not.
